src/Microsoft.DotNet.XHarness.Android/Execution/AdbProcessManager.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.XHarness.Android.Execution
+{
+    public class AdbProcessManager : IAdbProcessManager
+    {
+        private readonly ILogger log;
+        public AdbProcessManager(ILogger logger) => log = logger;
+
+        /// <summary>
+        ///  Whenever there are multiple devices attached to a system, most ADB commands will fail
+        ///  unless the specific device id is provided with -s {device serial #}
+        /// </summary>
+        public string DeviceSerial { get; set; } = "";
+
+        public ProcessExecutionResults Run(string adbExePath, string arguments)
+        { 
+            return Run(adbExePath, arguments, TimeSpan.FromMinutes(5));
+        }
+
+        public ProcessExecutionResults Run(string adbExePath, string arguments, TimeSpan timeOut)
+        {
+            string deviceSerialArgs = string.IsNullOrEmpty(DeviceSerial) ? string.Empty : $"-s {DeviceSerial}";
```

Comment:
I don't see why not make `DeviceSerial` nullable as well since now we still keep juggling with the empty strings (like in the AdbRunner `_processManager.DeviceSerial = deviceSerialNumber ?? string.Empty;`). Instead, `null` would just mean not set and we could react to it here accordingly

Reply:
Will consider it for later.  In this line whether it's empty or null I have to make the same check so I don't see the benefit.